addon/src/factory-guy.js

```diff
@@ -489,6 +488,8 @@ class FactoryGuy {
   }
 
   reset() {
+    this.store = null;
+    this.fixtureBuilderFactory = null;
     this.resetDefinitions();
     this.resetMockAjax();
```

Comment:
Rather than hooking into store.willDestroy to clear the store - we instead clear the store on each reset() call. This seems to work fine.